Stack/Core/Stack/State/NodeState.cs

```diff
@@ -2202,13 +2247,21 @@ public void SetAreEventsMonitored(ISystemContext context, bool areEventsMonitore
                 }
 
                 // propagate monitoring flag to target notifiers.
-                if (m_notifiers != null)
+                List<Notifier> notifierList = null;
```

Comment:
We should fix this so that we do not defensively copy the m_notifiers here or elsewhere where we are just reading it. Instead we should do the copying when the list is mutated eg when notifiers are added to the list. This avoids excessive copying when events are fired. The mutation of the list is much more rare than the events being forwarded. We could also change the list into an array.